File: src/main/java/br/com/projetojupiter/model/Conteudo.java
package br.com.projetojupiter.model;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@Entity
@Table(name="tb_conteudos")
public class Conteudo {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	
	@NotNull
	@Size(min=5, max=100)
	private String titulo;
	
	@NotNull
	@Size(min=10, max=200)
	private String descricao;
	
	@NotNull
	private String url;
	
	private Long ordemConteudo;
	
	@ManyToOne
	@JsonIgnoreProperties("conteudo")
	private Curso curso;
	

	public Curso getCurso() {
		return curso;
	}

	public void setCurso(Curso curso) {
		this.curso = curso;
	}

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getTitulo() {
		return titulo;
	}

	public void setTitulo(String titulo) {
		this.titulo = titulo;
	}

	public String getDescricao() {
		return descricao;
	}

	public void setDescricao(String descricao) {
		this.descricao = descricao;
	}

	public String getUrl() {
		return url;
	}

	public void setUrl(String url) {
		this.url = url;
	}

	public Long getOrdemConteudo() {
		return ordemConteudo;
	}

	public void setOrdemConteudo(Long ordemConteudo) {
		this.ordemConteudo = ordemConteudo;
	}
}

File: src/main/java/br/com/projetojupiter/model/Conteudo.kt
package br.com.projetojupiter.model

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import javax.persistence.*
import javax.validation.constraints.NotNull
import javax.validation.constraints.Size

@Entity
@Table(name = "tb_conteudos")
class Conteudo {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0

    @NotNull
    @Size(min = 5, max = 100)
    var titulo: String = ""

    @NotNull
    @Size(min = 10, max = 200)
    var descricao: String = ""

    @NotNull
    var url: String = ""

    var ordemConteudo: Long = 0

    @ManyToOne
    @JsonIgnoreProperties("conteudo")
    var curso: Curso? = null
}